src/solvers/refinement/string_refinement.cpp

```diff
@@ -812,33 +812,34 @@ decision_proceduret::resultt string_refinementt::dec_solve()
   }
 
   const auto get = [this](const exprt &expr) { return this->get(expr); };
-  for(exprt axiom : generator.get_axioms())
+
+  for(string_constraintt constraint : generator.get_constraints())
```

Comment:
I'd rewrite that as `std::transform(constraints.begin(), constraints.end(), std::back_inserter(axioms.universal), []...)`

Also this for loop is missing `const` and reference (each constraint is unnecessarily copied).